src/configuration/configuration.ts

```diff
@@ -56,6 +56,11 @@ class ConfigurationClass {
     /* tslint:disable:forin */
     // Disable forin rule here as we make accessors enumerable.`
     for (const option in this) {
+      // Some options are internal states and should not be reinitialized
+      if (option === "hl") {
```

Comment:
I will move this flag to vimState.globalState instead of Configuration so that this can be removed